public/app/features/admin/AdminListUsersCtrl.ts

```diff
@@ -40,10 +43,53 @@ export default class AdminListUsersCtrl {
     this.getUsers();
   }
 
+  addUsersAuthLabels() {
+    this.users.forEach(user => {
+      const authModuleLabel = this.getAuthLabel(user);
+      const labelStyle = this.getAuthLabelStyle(authModuleLabel);
+      user.authLabel = authModuleLabel;
+      user.authLabelStyle = labelStyle;
+    });
+  }
+
   getAuthModule(user: any) {
     if (user.authModule && user.authModule.length) {
       return user.authModule[0];
     }
     return undefined;
   }
+
+  getAuthLabel(user: any) {
+    const authModule = this.getAuthModule(user);
+
+    switch (authModule) {
```

Comment:
It can be questioned/discussed whether the raw auth module should be returned from the API. It doesn't really describe the intent/meaning, i.e. one of the authentication types the user sometimes have used. Guess this is something we can fix eventually.

To me it would make more sense to show all badges for one user when editing a user. Thought?

Reply:
Just saw that you have this exact same login in backend now since this is merged to master: https://github.com/grafana/grafana/pull/17860/files#diff-1c4af4bbe7e52b7dc3d14888a6c42459R33

Then I would like us to not duplicate this code and let the APO return labels.

Reply:
yea, that looks bad to have that duplicated.

Reply:
Yes, I can rewrite it now.

Reply:
Great :+1: